lib/ansible/modules/crypto/luks_device.py

```diff
@@ -251,15 +317,22 @@ def is_luks(self, device):
 
     def run_luks_create(self, device, keyfile):
         # create a new luks container; use batch mode to auto confirm
-        result = self._run_command([self._cryptsetup_bin, 'luksFormat',
-                                    '-q', device, keyfile])
+        label = self._module.params['label']
+        if label is not None:
+            # create luks container v2 with label
+            result = self._run_command([self._cryptsetup_bin, 'luksFormat',
+                                        '--type', 'luks2', '--label', label,
+                                        '-q', device, keyfile])
+        else:
+            result = self._run_command([self._cryptsetup_bin, 'luksFormat',
+                                        '-q', device, keyfile])
         if result[RETURN_CODE] != 0:
             raise ValueError('Error while creating LUKS on %s: %s'
                              % (device, result[STDERR]))
 
     def run_luks_open(self, device, keyfile, name):
         result = self._run_command([self._cryptsetup_bin, '--key-file', keyfile,
-                                    'open', '--type', 'luks', device, name])
```

Comment:
Is there a reason why you want to remove this? Does this improve compatibility in any way, or is required for LUKS2 support?

Reply:
Its required for LUKS2 support, if pass `--type luks` it will make LUKS1 container on some versions.

Reply:
According to the man page, `luks` (the default value) opens both LUKS1 and LUKS2 containers; see here: http://man7.org/linux/man-pages/man8/cryptsetup.8.html#BASIC_COMMANDS

Do you know of a system where `--type luks` will not open LUKS2 containers? This is about opening, not creating.

Reply:
You are right sir, my bad!

Reply:
No problem. In fact, removing it shouldn't hurt, since `luks` is the default. But I guess it's better to keep it, just in case somewhen in the future, the default changes.